MuonAnalysis/MuonAssociators/plugins/MuonFSRAssociator.cc

```diff
@@ -0,0 +1,89 @@
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/global/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/Candidate/interface/Candidate.h"
+#include "DataFormats/PatCandidates/interface/GenericParticle.h"
+#include "DataFormats/Math/interface/LorentzVector.h"
+
+#include "DataFormats/PatCandidates/interface/Muon.h"
+#include "DataFormats/Common/interface/ValueMap.h"
+
+#include "DataFormats/Common/interface/Association.h"
+
+//
+// class declaration
+//
+
+class MuonFSRAssociator : public edm::global::EDProducer<> {
+public:
+  explicit MuonFSRAssociator(const edm::ParameterSet& iConfig)
+      :
+
+        photons_{consumes<pat::GenericParticleCollection>(iConfig.getParameter<edm::InputTag>("photons"))},
+        muons_{consumes<edm::View<reco::Muon>>(iConfig.getParameter<edm::InputTag>("muons"))}
+
+  {
+    produces<edm::Association<std::vector<pat::GenericParticle>>>();
+    produces<edm::ValueMap<int>>("fsrIndex");
+  }
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+    desc.add<edm::InputTag>("photons")->setComment("FSR photon collection to associate with muons");
+    desc.add<edm::InputTag>("muons")->setComment("collection of muons to associate with FSR photons");
+
+    descriptions.add("MuonFSRAssociator", desc);
```

Comment:
CMS style for a default module name is to start from a lower case ```muonFSRAssociator```
or can just use ```descriptions.addWithDefaultLabel(desc);```